### PR TITLE
Add back a log message for when the network is successfully established

### PR DESCRIPTION
**Pull request by umanwizard**
These messages got lost in the refactor.

### Motivation

  * This PR fixes a previously unreported bug.

    It is no longer easy to tell from logs whether the Timely network has finished setting up 

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

(None)

### CODE REVIEW

File: src/compute/src/communication.rs [hunk 1]
```diff
@@ -82,11 +82,15 @@ pub async fn initialize_networking(
...
-        )),
+        Ok((stuff, guard)) => {
+            info!(process = process, "successfully initialized network");
+            Ok((
+                stuff.into_iter().map(GenericBuilder::ZeroCopy).collect(),
+                Box::new(guard),
+            ))
+        }
         Err(err) => {
+            error!(process = process, "failed to initialize network: {err}");
```
Comment (benesch) [inline src/compute/src/communication.rs:93]: `error!` will report to Sentry. Is that the right thing to use here? We're trying to limit `error!`s to stuff that will one day page the on call engineer.

Comment (umanwizard) [reply src/compute/src/communication.rs:93]: no, it can be a warning

Comment (umanwizard) [inline src/compute/src/communication.rs:93]:
```suggestion
            warn!(process = process, "failed to initialize network: {err}");
```

**Review by benesch — APPROVED** [review]

File: src/compute/src/communication.rs [hunk 2]
```diff
@@ -46,7 +46,7 @@ use mz_compute_client::command::ComputeStartupEpoch;
 use mz_ore::cast::CastFrom;
 use timely::communication::allocator::zero_copy::initialize::initialize_networking_from_sockets;
 use timely::communication::allocator::GenericBuilder;
-use tracing::{debug, info, warn};
+use tracing::{debug, error, info, warn};
```
Comment (umanwizard) [inline src/compute/src/communication.rs:49]:
```suggestion
use tracing::{debug, info, warn};
```

File: src/compute/src/communication.rs [hunk 3]
```diff
@@ -82,11 +82,15 @@ pub async fn initialize_networking(
...
-        )),
+        Ok((stuff, guard)) => {
+            info!(process = process, "successfully initialized network");
+            Ok((
+                stuff.into_iter().map(GenericBuilder::ZeroCopy).collect(),
+                Box::new(guard),
+            ))
+        }
         Err(err) => {
+            warn!(process = process, "failed to initialize network: {err}");
```
Comment (teskje) [inline src/compute/src/communication.rs:93]:
```suggestion
            warn!(process, "failed to initialize network: {err}");
```
I *think* this works as well? But just a nit in any case.

**Review by teskje — APPROVED** [review]

**Review by lluki — APPROVED** [review]
LGTM

**Pull request merged**